libs/patchwheel/parse.go

```diff
@@ -0,0 +1,58 @@
+package patchwheel
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// WheelInfo contains information extracted from a wheel filename
+type WheelInfo struct {
+	Distribution string   // Package distribution name
+	Version      string   // Package version
+	Tags         []string // Python tags (python_tag, abi_tag, platform_tag)
+}
+
+// calculateNewVersion generates a new version string and filename based on the wheel info and modification time.
+// The version is updated according to the following rules:
+//   - if there is an existing part after + it is dropped
+//   - append +<mtime of the original wheel> to version
+func calculateNewVersion(info WheelInfo, mtime time.Time) (newVersion, newFilename string) {
+	baseVersion := strings.SplitN(info.Version, "+", 2)[0]
+
+	dt := strings.Replace(mtime.Format("20060102150405.00"), ".", "", 1)
+	dt = strings.Replace(dt, ".", "", 1)
```

Comment:
You want to remove two `.`'s here?

It would be good to include a comment saying what the input/output shape is.

Reply:
I assume the reason to include fractional seconds is to make it "more unique"?

That's also the reason why you need to remove the decimal point in the output string.

If this is the case, this stanza can use its own comment to clarify this.

Reply:
Btw, if the number of decimals controls the resolution, I think we should use 3 for millis.